File: doordeck-sdk/src/mingwMain/resources/doordeck_headless_sdk/model/responses/SiteResponses.cs
namespace DoordeckHeadlessSDK.model.responses
{
    public class SiteResponse
    {
        public string id { get; set; }
        public string name { get; set; }
        public string colour { get; set; }
        public double longitude { get; set; }
        public double latitude { get; set; }
        public int radius { get; set; }
        public string passBackground { get; set; }
        public string created { get; set; }
        public string updated { get; set; }
    }

    public class SiteLocksResponse
    {
        public string id { get; set; }
        public string name { get; set; }
        public string? colour { get; set; } = null;
        public UserRole role { get; set; }
        public SiteLockSettingsResponse settings { get; set; }
    }

    public class SiteLockSettingsResponse
    {
        public double unlockTime { get; set; }
        public List<string> permittedAddresses { get; set; }
        public string defaultName { get; set; }
        public List<string> tiles { get; set; }
        public SiteStateResponse? state { get; set; } = null;
        public bool? favourite { get; set; } = null;
    }

    public class SiteStateResponse
    {
        public bool connected { get; set; }
    }

    public class UserForSiteResponse
    {
        public string userId { get; set; }
        public string email { get; set; }
        public string? displayName { get; set; } = null;
        public bool orphan { get; set; }
    }
}

File: doordeck-sdk/src/commonMain/kotlin/com/doordeck/multiplatform/sdk/api/responses/SiteResponses.kt
package com.doordeck.multiplatform.sdk.api.responses

import com.doordeck.multiplatform.sdk.api.model.UserRole
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
data class SiteResponse(
    val id: String,
    val name: String,
    val colour: String,
    val longitude: Double,
    val latitude: Double,
    val radius: Int,
    val passBackground: String,
    val created: String,
    val updated: String
)

@JsExport
@Serializable
data class SiteLocksResponse(
    val id: String,
    val name: String,
    val colour: String? = null,
    val role: UserRole,
    val settings: SiteLockSettingsResponse
)

@JsExport
@Serializable
data class SiteLockSettingsResponse(
    val unlockTime: Double,
    val permittedAddresses: List<String>,
    val defaultName: String,
    val tiles: List<String>,
    val state: SiteStateResponse? = null,
    val favourite: Boolean? = null
)

@JsExport
@Serializable
data class SiteStateResponse(
    val connected: Boolean
)

@JsExport
@Serializable
data class UserForSiteResponse(
    val userId: String,
    val email: String,
    val displayName: String? = null,
    val orphan: Boolean
)

